pkg/operator/ceph/cluster/osd/volumes.go

```diff
@@ -86,7 +87,26 @@ func getDeviceMapperVolume() (v1.Volume, v1.VolumeMount) {
 	return volume, volumeMounts
 }
 
-func getPVCOSDVolumes(osdProps *osdProperties) []v1.Volume {
+func getDataBridgeVolumeSource(claimName, configDir, namespace string, inProvisioning bool) v1.VolumeSource {
+	var source v1.VolumeSource
+	if inProvisioning {
+		source.EmptyDir = &v1.EmptyDirVolumeSource{
+			Medium: "Memory",
+		}
+	} else {
+		hostPathType := v1.HostPathDirectoryOrCreate
```

Comment:
The purpose of the hostpath is to prevent multiple OSD pods from launching the same OSD and causing corruption, right? How about a comment about why we need the hostpath and how ceph uses it to ensure the OSD is protected?

Reply:
Indeed, I'll add it.

Reply:
What is the behavior when the lock is held? How about adding more comments in the code as well to answer these?
- The second pod fails to start and k8s will restart the second osd pod until the first one releases the volume? 
- If the first pod is terminated abruptly (instead of a graceful lockdown), the lock will always be released, right?

Since Rook isn't implementing the lock I'm sure the behavior is fine, I just want to confirm that the Ceph lock works well for Rook scenarios.

Reply:
The lock is released once the process is gone, so until the ceph-osd is alive, the other pod (same OSD) will not be able to acquire it. That's actually the fix.

Reply:
Add more comments, thanks.